data/campaigns/tutorial/scenarios/01_Tutorial_part_1.cfg

```diff
@@ -194,6 +210,85 @@
         [/lua]
     [/event]
 
+    # Give a warning to undo last step if target is out of reach
+    # Check whether at least one of our units can reach any of the target_loc locations
+    [event]
+        name=moveto
+        id=check_undo
+        first_time_only=no
+        [filter]
+            side=1
+        [/filter]
+        [allow_undo][/allow_undo]
+
+        # Store all our units
+        [store_unit]
+            [filter]
+                side=1
+            [/filter]
+            variable=student_team
+        [/store_unit]
+
+        {VARIABLE within_reach no}
+        #{VARIABLE mindist 100} # kept for debugging
+
+        [foreach]   # Check each unit
+            array=student_team
+            [do]
+                {VARIABLE moves_left $this_item.moves}
+                {VARIABLE unit_id $this_item.id}
+
+                [foreach]  # compare their movement points with the movement costs of the valid target locations
+                    array=target_loc
+                    [do]
+                        [find_path]
+                            [traveler]
+                                id=$unit_id
```

Comment:
Please rename this variable, it looks to similar to $unit.id, which is a variable which is offered by the engine (the unit which did trigger this event)